### PR TITLE
[Core] Add return types to Module class

### PR DESCRIPTION
**Pull request by johnsaigle**
### Brief summary of changes
Add return types to the Module class

### CODE REVIEW

File: php/libraries/Module.class.inc [hunk 1]
```diff
@@ -150,7 +153,7 @@ class Module extends \LORIS\Router\PrefixRouter implements RequestHandlerInterfa
      * Returns true if this module should be publicly available to people who
      * are not logged in to LORIS.
      *
-     * @return bool (false: not that this is not currently possible because main.php
+     * @return bool (false: note that this is not currently possible because main.php
      *         checks that the user is logged in before it calls NDB_Caller.)
      */
     public function isPublicModule()
```
Comment (johnsaigle) [inline php/libraries/Module.class.inc]: I wasn't able to add a return type here without offending `phan`. Thsi signature must match the function in `modules/api/php/module.class.inc` but when both are delcared to return bool I get the following error.
```
modules/login/php/module.class.inc:36 PhanParamSignatureRealMismatchReturnType Declaration of function isPublicModule() should be compatible with function isPublicModule() : bool (method returning '' cannot override method returning 'bool') defined in php/libraries/Module.class.inc:159
```

I wasn't able to fix it.

Comment (driusan) [reply php/libraries/Module.class.inc]: That sounds like the login module class (and likely all the module's module classes) needs to be updated to have the same signature as the parent class.

File: php/libraries/Module.class.inc [hunk 2]
```diff
@@ -150,7 +154,7 @@ class Module extends \LORIS\Router\PrefixRouter implements RequestHandlerInterfa
      * Returns true if this module should be publicly available to people who
      * are not logged in to LORIS.
      *
-     * @return bool (false: not that this is not currently possible because main.php
+     * @return bool (false: note that this is not currently possible because main.php
```
Comment (driusan) [inline php/libraries/Module.class.inc]: I know you're just fixing a typo, but I don't think this note is relevant anymore (main.php is only used for instrument loading on major right now.)

Comment (johnsaigle) [reply php/libraries/Module.class.inc]: Should I delete it?

Comment (driusan) [reply php/libraries/Module.class.inc]: Might as well since you're changing it anyways, otherwise we'll have to do it later

**Comment by zaliqarosli** [comment]
@johnsaigle please rebase to include https://github.com/aces/Loris/pull/3897 for testing

**Pull request merged**